### PR TITLE
Update chowder.py

### PR DESCRIPTION
**Pull request by VolodymyrChapman**
Add metadata_cols argument to Chowder class to allow for input of data with preceding columns (for example, magnification) to be ignored by the chowder model. 
Default of 3 assumes magnification, start_x and start_y columns preceding data to be fitted / inferred on, in keeping with current behaviour of 3 preceding columns - i.e. default settings will not break behaviour of existing Chowder models / code.
Testing on fresh conda environment with modified Chowder model:

image

### CODE REVIEW

**Comment by jbschiratti** [comment]
Thanks @VolodymyrChapman for initiating this PR.

To keep a consistent behavior, can you also add the `metadata_cols` parameter in other [MIL models](https://github.com/owkin/HistoSSLscaling/tree/main/rl_benchmarks/models/slide_models)?

**Comment by VolodymyrChapman** [comment]
@jbschiratti Of course - will do :-)

**Comment by jbschiratti** [comment]
Thanks @VolodymyrChapman!

Although there's no CI (yet) on this repository, can you, please, update the unit tests and make sure they pass locally?

**Comment by jbschiratti** [comment]
@VolodymyrChapman Looks like you pushed some unnecessary `.pyc` files by mistake. Can you please remove them from this PR?

**Comment by VolodymyrChapman** [comment]
@jbschiratti Of course :-)
The metadata_cols variable has been added to all MIL models, as per the Chowder model.
The variable has also been added to all models in the tests/models/test_slide_models.py file.
Linting was performed with black using the specifications outlined in dev_tools/linting.sh
Local output of black checks and unittest for just the MIL models (Ubuntu 22.04 in a fresh conda env):
image

Are there any other tests / checks that you would require?

**Comment by jbschiratti** [comment]
@VolodymyrChapman Once you've removed the `.pyc` files, this will be good to go. 

I'll let @afilt take over as he's the code owner on this repository.

**Comment by VolodymyrChapman** [comment]
@jbschiratti  Thank you for spotting - apologies for the spam files!
I think they've all been removed now.

File: rl_benchmarks/models/slide_models/abmil.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ class ABMIL(nn.Module):
         Activation for last MLP.
     bias: bool = True
         Add bias to the first MLP.
+    metadata_cols: int = 3
+        Number of metadata columns (for example, magnification, patch start
+        coordinates etc.) at the start of input data. Default of 3 assumes
+        magnification, patch start x and patch start y.
```
Comment (afilt) [inline rl_benchmarks/models/slide_models/abmil.py]: Minor rephrasing, could you replace by smgth more specific: "Default of 3 assumes the deep zoom level (corresponding to a given magnification), x and y coordinates of the patch." ? Thanks a lot !

File: rl_benchmarks/models/slide_models/chowder.py [hunk 2]
```diff
@@ -54,6 +54,10 @@ class Chowder(nn.Module):
         Activation that is used after each layer of the MLP.
     bias: bool = True
         Whether to add bias for layers of the tiles MLP.
+    metadata_cols: int = 3
+        Number of metadata columns (for example, magnification, patch start
+        coordinates etc.) at the start of input data. Default of 3 assumes
+        magnification, patch start x and patch start y.
```
Comment (afilt) [inline rl_benchmarks/models/slide_models/chowder.py]: Same applies here.

File: rl_benchmarks/models/slide_models/abmil.py [hunk 3]
```diff
@@ -138,7 +145,7 @@ def forward(
         logits, attention_weights: Tuple[torch.Tensor, torch.Tensor]
             (B, OUT_FEATURES), (B, N_TILES)
         """
-        tiles_emb = self.tiles_emb(features[..., 3:], mask)
+        tiles_emb = self.tiles_emb(features[..., self.metadata_cols :], mask)
```
Comment (afilt) [inline rl_benchmarks/models/slide_models/abmil.py]: Could you please remove the blank space between `self.metadata_cols` and `:` ? You can apply black formatting using `black --line-length=79 --check --diff rl_benchmarks/models/slide_models/`. Thanks ! Same applies for the other files.

**Review by afilt — COMMENTED** [review]
Thanks a lot @VolodymyrChapman for adding this parameter to the MIL models. I left 2 minor comments that would require changes imo. Thanks @jbschiratti for the follow-up !

**Review by afilt — APPROVED** [review]
Looks good to me, thanks a lot @VolodymyrChapman 🚀 Can you merge and delete your branch ? Thanks.

File: rl_benchmarks/models/slide_models/abmil.py (hunk 1 above)
Comment (VolodymyrChapman) [reply rl_benchmarks/models/slide_models/abmil.py]: Hi @afilt - my pleasure.
Thank you for the feedback - I believe this has been actioned now. Please let me know whether the explanations are satisfactory now. :-)

File: rl_benchmarks/models/slide_models/chowder.py (hunk 2 above)
Comment (VolodymyrChapman) [reply rl_benchmarks/models/slide_models/chowder.py]: As previous comment :-)

File: rl_benchmarks/models/slide_models/abmil.py (hunk 3 above)
Comment (VolodymyrChapman) [reply rl_benchmarks/models/slide_models/abmil.py]: Blank spaces manually removed :-)
FYI - I believe that it is black formatting that added this space as the --diff output suggests addition of it:
image

**Comment by VolodymyrChapman** [comment]
@afilt Thank you! I don't think I have necessary write access to perform the merge as the button is missing from my view:
image
Please let me know if the merge is successful from your end so I can delete the fork :-)

**Comment by afilt** [comment]
You're right :-) Thanks a lot @VolodymyrChapman !

**Comment by VolodymyrChapman** [comment]
My pleasure - happy to help :-)

**Pull request merged**